scalahost/sbt/src/main/scala/scala/meta/scalahost/sbt/ScalahostSbtPlugin.scala

```diff
@@ -22,15 +40,21 @@ object ScalahostSbtPlugin extends AutoPlugin {
 
   lazy val scalahostAllSettings =
     scalahostBaseSettings ++
-      scalahostInjectCompilerPluginSettings ++
+      inTestAndCompile(scalahostConfigSettings) ++
+      scalahostInjectScalacOptions ++
+      scalahostInjectDependency ++
       scalahostHasMirrorSettings
 
   lazy val scalahostBaseSettings: Seq[Def.Setting[_]] = Def.settings(
     ivyConfigurations += Scalameta,
+    scalametaSourcepath := baseDirectory.in(ThisBuild).value,
     resolvers += Resolver.bintrayRepo("scalameta", "maven")
   )
+  lazy val scalahostConfigSettings: Seq[Def.Setting[_]] = Def.settings(
+    scalametaSemanticdb := ScalametaSemanticdb.Fat
```

Comment:
Good point. How about we explore this in 2.x? I'm afraid big refactorings like this will interfere with the 2.x branch, which will have to coexist with 1.x for quite a while.